sample_formatted_code.do_not_merge.c

```diff
@@ -0,0 +1,562 @@
+
+    // Same as above but with additional parameter using ContinuationIndentWith
+    SCLogInfo("running in 'auto' checksum mode. Detection of interface "
+              "state will require " xstr(CHECKSUM_SAMPLE_COUNT) " packets %d",
+            someValue);
+
+    // Just params use ContinuationIndentWith
+    SCLogError(SC_ERR_STAT, "(%s) Failed to get pcap_stats: %s", tv->name,
+            pcap_geterr(ptv->pcap_handle));
+
+    // string literal param that fits on continuation line uses
+    // ContinuationIndentWith
+    SCLogError(SC_ERR_INITIALIZATION,
+            "Error getting context for Prelude. \"initdata\" argument NULL");
+    if (unlikely(initdata == NULL)) {
+        // string literal param does not fit on continuation line
+        // Continuation is from "string literal start", not
+        // ContinuationIndentWith, due to reasons
+        SCLogError(SC_ERR_INITIALIZATION, "Error getting context for Prelude.  "
+                                          "\"initdata\" argument NULL");
+        SCReturnInt(TM_ECODE_FAILED);
+    }
+
+    // function call inside if needs breaking apart and uses
+    // ContinuationIndentWith starting at function. Nice.
+    if (DetectContentDataParse("fileext", str, &fileext->ext, &fileext->len,
+                &fileext->flags) == -1) {
+        goto error;
+    }
+}
+
+//--- foreach handling as "for loop" ---
+void foreach_handling()
+{
+    // ForEachMacros
+
+    // json_object_foreach and json_array_foreach are "foreach" functions
+    if (json_is_object(value)) {
+        json_object_foreach (value, key_js, value_js) {
+            bla()
+        }
+    } else if (json_is_array(value)) {
+        json_array_foreach (value, index, value_js) {
+            bla()
+        }
+    } else if (json_is_integer(value)) {
+        ret = AddIntData(alert, key, json_integer_value(value));
+    }
+
+    // These are foreach macros but apart from the odd exception, they
+    // use start parens on next line if they are used in code.
+    // It would be trivial to also handle them like "for loops"
+
+    // tree.h: SLIST_FOREACH, SLIST_FOREACH_PREVPTR, LIST_FOREACH,
+    // SIMPLEQ_FOREACH, TAILQ_FOREACH, TAILQ_FOREACH_SAFE,
+    // TAILQ_FOREACH_REVERSE, CIRCLEQ_FOREACH, CIRCLEQ_FOREACH_REVERSE,
+    // CIRCLEQ_FOREACH_SAFE, CIRCLEQ_FOREACH_REVERSE_SAFE
+
+    TAILQ_FOREACH(child, &node->head, next)
+    {
+        name[level] = SCStrdup(child->name);
+        /* ... */
+        SCFree(name[level]);
+    }
+
+    // queue.h: SPLAY_FOREACH, RB_FOREACH, RB_FOREACH_FROM, RB_FOREACH_SAFE,
+    // RB_FOREACH_REVERSE, RB_FOREACH_REVERSE_FROM, RB_FOREACH_REVERSE_SAFE
+    RB_FOREACH_REVERSE_FROM(tree_seg, TCPSEG, s)
+    {
+        if (tree_seg == seg)
+            continue;
+        /* ... */
+    }
+}
+
+//--- comment wrapping ---
+void multi_line_comments()
+{
+    // ReflowComments: false does not trim comments to ColumnLimit chars
```

Comment:
I disabled limiting comments to 80 chars in the config. Hence, clang-format does not reformat comments.

While it's great to use in my experience, changing an existing code base can be messy. It requires manual fixing up of comments as we devs suck at formatting comments to a length.

Having said that, it probably would be great to enable it so that new code could take advantage of it.

Reply:
I really like it when an IDE reflows my comments to 80 chars.  So I'd like to see this on by default, if it doesn't get in the way too often.